common/authorization/oauthAuthorizer.go

```diff
@@ -101,49 +104,63 @@ func (a *oauthAuthority) Authorize(
 		return Result{Decision: DecisionDeny}, nil
 	}
 
-	claims, err := a.parseToken(token, a.verifier)
+	var claims JWTClaims
+
+	_, err := a.parser.ParseWithClaims(token, &claims, a.keyFunc)
+
 	if err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
-	if err := a.validateTTL(claims); err != nil {
+	if err := a.validateTTL(&claims); err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
 	if claims.Admin {
 		return Result{Decision: DecisionAllow}, nil
 	}
+
 	domain, err := a.domainCache.GetDomain(attributes.DomainName)
 	if err != nil {
 		return Result{Decision: DecisionDeny}, err
 	}
 
-	if err := validatePermission(claims, attributes, domain.GetInfo().Data); err != nil {
+	if err := validatePermission(&claims, attributes, domain.GetInfo().Data); err != nil {
 		a.log.Debug("request is not authorized", tag.Error(err))
 		return Result{Decision: DecisionDeny}, nil
 	}
 
 	return Result{Decision: DecisionAllow}, nil
 }
 
-func (a *oauthAuthority) parseToken(tokenStr string, verifier jwt.Verifier) (*JWTClaims, error) {
-	token, err := jwt.ParseAndVerifyString(tokenStr, verifier)
-	if err != nil {
-		return nil, fmt.Errorf("parse token: %w", err)
-	}
-	var claims JWTClaims
-	_ = json.Unmarshal(token.RawClaims(), &claims)
-	return &claims, nil
+// keyFunc returns correct key to check signature
+func (a *oauthAuthority) keyFunc(token *jwt.Token) (interface{}, error) {
+	// only local public key is supported currently
+	return a.publicKey, nil
 }
 
 func (a *oauthAuthority) validateTTL(claims *JWTClaims) error {
-	if claims.TTL > a.authorizationCfg.MaxJwtTTL {
-		return fmt.Errorf("token TTL: %d is larger than MaxTTL allowed: %d", claims.TTL, a.authorizationCfg.MaxJwtTTL)
+	// Fill ExpiresAt when TTL is passed
+	if claims.TTL > 0 {
+		claims.ExpiresAt = jwt.NewNumericDate(claims.IssuedAt.Time.Add(time.Second * time.Duration(claims.TTL)))
 	}
-	if claims.Iat+claims.TTL < time.Now().Unix() {
-		return errors.New("JWT has expired")
+
+	exp, err := claims.GetExpirationTime()
+
+	if err != nil || exp == nil {
+		return errors.New("ExpiresAt is not set")
 	}
+
+	timeLeft := exp.Unix() - time.Now().Unix()
```

Comment:
my comment wasn't great, I actually meant this: I thought this would be something that the library would do natively